torchtext/experimental/vectors.py

```diff
@@ -0,0 +1,38 @@
+import csv
+import torch
+
+
+def vectors_from_csv_file(file_like_object, unk_tensor=None):
+    r"""Create a Vectors object from a csv file like object.
+
+    Args:
+        file_like_object (FileObject): a file like object to read data from.
+        unk_tensor (int): a 1d tensors representing the vector associated with an unknown token
+    Returns:
+        Vectors: a Vectors object.
+    """
+    readCSV = csv.reader(file_like_object, delimiter=',')
+
+    tuples = list(map(lambda x: (x[0], torch.tensor([float(c) for c in x[1].split()], dtype=torch.float)), readCSV))
```

Comment:
It should be better to just do a for-loop and append. The code I wrote below was meant of brevity (where map probably actually didn't help).

The reason for that is, that within the current code you're creating two lists of length equal to the file length instead of just one.